src/main/java/pl/arturzgodka/controllers/SkillsController.java

```diff
@@ -24,13 +31,34 @@ public String getHeroesList(Model model) {
     @RequestMapping("skills/{heroClasses}")
     public String getSkills(Model model, @PathVariable(value="heroClasses") String heroClass) {
 
-        String heroClassWithHyphonSeparator = heroClass.replace(" ", "-");
-        List<HeroSkillDataModel> skillsMapped = skillMapper.mapSkillsToDataModel(heroClassWithHyphonSeparator, getHeroClassSkillsList(heroClassWithHyphonSeparator)); //z innej klasy nie moge tak zrobic bo dostaję błąd, server status 500.
+        List<HeroSkillDataModel> skillsMapped = getSkillsToDisplay(heroClass);
 
         model.addAttribute("skills", skillsMapped);
         model.addAttribute("heroClass", heroClass);
 
         return "skills";
     }
 
+    @GetMapping("skills/skillSearch")
+    public String searchSkillForm(Model model, @RequestParam String skillSearchName, @RequestParam String heroClassName) {
+
+        List<HeroSkillDataModel> skillsMatched = getSkillsToDisplay(heroClassName, skillSearchName);
+
+        model.addAttribute("skills", skillsMatched);
+        model.addAttribute("heroClass", heroClassName);
+        model.addAttribute("skillSearched", true); //flag to not display search bar after search was submitted and results are being displayed
+
+        return "skills";
+    }
+
+    private List<HeroSkillDataModel> getSkillsToDisplay(String heroClassName, String skillSearchName) {
```

Comment:
Should this really be in a controller?

Reply:
It is additional method, I moved it out from controller to seperate class just for searching functionality: SearchSkill class.